.github/workflows/ci.yml

```diff
@@ -204,7 +216,11 @@ jobs:
           echo "::warning::CMake: $(cmake --version)"
           echo "::warning::Compiler: $(${{ matrix.cxx_compiler }} --version)"
 
-          if [[ "${{ matrix.backend }}" == "dpcpp" ]]; then
+          if [[ "${{ matrix.std }}" == "23" ]]; then
```

Comment:
Eventually, oneDPL main will get to std 23 (it is probably a somewhat distant future...).  Perhaps we can worry about this then, but would it be better to just have a distributed ranges only flag in the matrix rather than making this trigger off of the standard library version?

Generally, its a little strange having an experimental feature which requires std 23 in a library that doesn't officially support std 23 otherwise.  I understand that distributed ranges depends upon components of oneDPL internally.  Presumably the pieces of oneDPL which are used internally support std 23 now, otherwise you couldn't use them successfully.  Are there selective tests we can turn on from oneDPL with this in mind, possibly async tests or some subset?

Reply:
In https://github.com/oneapi-src/oneDPL/pull/1691 I've changed CI in a way that the same tests which oneDPL runs on dpcpp backend are also executed in 23 standard. Luckily they all pass, see [this run](https://github.com/oneapi-src/oneDPL/actions/runs/9885952177/job/27304866796?pr=1691). Additionally DR tests are run in this configuration.

My advise is not to add another flag to matrix. "If" inside the code better expresses what conditions for running several tests are and do not complicate the matrix.

Regarding enabling (selected) oneDPL tests in 23 standard:
This PR already enables them in per-commit CI. We have a task to add DR tests to your nightly PVC suite. The task is about to try add also oneDPL tests testing functionality used by DR. This is however not so critical because - as you mentioned - pieces in oneDPL which we use are executed by DR tests and we know and already continuously verify that they work with 23 standard.

Reply:
I agree this isn't an urgent requirement to run the oneDPL tests along with the DR tests in the per-commit CI.  I mention it because when something does fail, it will be better to be able to locate where the root cause of the error is from the test results.  If we tests DR and also the relevant oneDPL tests, that will give us a better idea, but it is more important in the full nightly runs, not the per-commit CI.
Eventually, we will want the logic to "add in" DR tests when standard 23, but keep the rest of the testing, but that can wait until later. We can call this resolved here.